src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1475 @@
+        dtype (optional): Defaults to torch.float.
+        eps (float, optional): The epsilon used by the layer normalization layers.
+        dropout_p (float, optional): Defaults to 0.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        num_layers: int = 48,
+        dim_model: int = 4096,
+        dim_ff: int = 10240,
+        num_heads: int = 32,
+        dim_head: int = 128,
+        dtype: torch.dtype = torch.float,
+        eps: float = 1e-6,
+        dropout_p: Optional[float] = 0.0,
+        mask_modules: Optional[List[Tuple[bool, bool]]] = None,
+    ):
+
+        super().__init__()
+        self.num_layers = num_layers
+
+        if mask_modules is not None:
+            # assert len(mask_modules) == num_layers, "The total number of masks should equal to num_layers"
+            if len(mask_modules) == num_layers:
+                raise ValueError("The total number of masks should equal to num_layers")
+            for mask_module in mask_modules:
+                # assert len(mask_module) == 2, "For encoder, each mask should be (mask_att, mask_ffn)"
+                if len(mask_module) != 2:
+                    raise ValueError("For encoder, each mask should be (mask_att, mask_ffn)")
+        else:
+            mask_modules = [(False, False)] * num_layers
+
+        self.layers = nn.ModuleList(
+            [
+                CPMAntTransformerBlock(
+                    dim_model=dim_model,
+                    dim_ff=dim_ff,
+                    num_heads=num_heads,
+                    dim_head=dim_head,
+                    dtype=dtype,
+                    eps=eps,
+                    dropout_p=dropout_p,
+                    mask_att=mask_modules[ith][0],
+                    mask_ffn=mask_modules[ith][1],
+                )
+                for ith in range(num_layers)
+            ]
+        )
+
+        self.output_layernorm = CPMAntLayerNorm(dim_norm=dim_model, dtype=dtype, eps=eps)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: torch.Tensor,
+        position_bias: torch.Tensor,
+        use_cache: bool = False,
+        past_key_values: Optional[List[Tuple[torch.Tensor, torch.Tensor]]] = None,
+    ):
+        """
+        Args:
+            hidden_states (`torch.Tensor` of `(batch, seq_enc, dim_model)`):
+                Input of encoder, might be the embedding of a batch of sequences.
+            attention_mask (`torch.Tensor` of `(batch, seq_enc, seq_enc)`):
+                Avoid invalid areas to participate in the calculation.
+            position_bias (`torch.Tensor` of shape `(num_heads, seq_enc, seq_enc)`):
+                Provides position information to attention mechanism.
+            use_cache (`bool`): Whether use cache.
+            past_key_values (`Tuple(torch.FloatTensor)`, *optional*): Cached past key and value projection states.
+        """  # noqa: E501
+        if not use_cache:
+            for layer in self.layers:
```

Comment:
can you upstream `use_cache` at the layer level? i.e. can you pass this argument in `CPMAntTransformerBlock` forward argument instead?